Directory.Build.props

```diff
@@ -21,6 +21,13 @@
     <BeforeTargetFrameworkInferenceTargets>$(RepositoryEngineeringDir)BeforeTargetFrameworkInference.targets</BeforeTargetFrameworkInferenceTargets>
     <!-- Only upgrade NuGetAudit warnings to errors for official builds. -->
     <WarningsNotAsErrors Condition="'$(OfficialBuild)' != 'true'">$(WarningsNotAsErrors);NU1901;NU1902;NU1903;NU1904</WarningsNotAsErrors>
+    <IsSourceProject Condition="'$(IsSourceProject)' == '' and
+                                '$(IsTestProject)' != 'true' and
+                                '$(UsingMicrosoftTraversalSdk)' != 'true'">true</IsSourceProject>
```

Comment:
Why did you move this property here? I don't see anything below conditioned on it.

Reply:
Before my last commit, I was using it here. Now I'm not, as you noticed. But that was not the only reason why I moved it here: I realized that all the 3 properties in the condition are available for use in the props file, just like IsTestProject, which is also available at this point. So I thought moving IsSourceProject to the props file was appropriate.

I could revert the change since it's not needed anymore, or we can keep it here since I think it's where it should be defined. I'm more inclined to keep it here.

Reply:
Yes, better keep it there.